src/service/viewer-impl.js

```diff
@@ -972,6 +972,12 @@ export class Viewer {
  * @private
  */
 function parseParams_(str, allParams) {
+  // Temporary workaround for a viewer sending a bad URL
+  // that has two # inside of it.
+  // See Google bug b/33706977
+  // If there is a # (but not at the start of the string)
+  // replace it with & before parsing.
+  str = str.replace(/(.)\#/, '$1&');
```

Comment:
I don't think `location.hash` is ever decoded. Most browsers don't encode it (but Firefox does ). Anyway, fixing this on the production side. Will close PR.